test/rql_test/connections/asyncio_connection.py

```diff
@@ -0,0 +1,923 @@
+            yield from cursor.next()
+            count += 1
+            if count == 2:
+                cursor.close()
+
+        self.assertTrue(count >= 2, "Did not get enough cursor results")
+
+    @asyncio.coroutine
+    def test_cursor_success(self):
+        range_size = 10000
+        cursor = yield from r.range().limit(range_size).run(self.conn)
+        count = 0
+        while (yield from cursor.fetch_next()):
+            yield from cursor.next()
+            count += 1
+        self.assertEqual(count, range_size,
+             "Expected %d results on the cursor, but got %d" % (range_size, count))
+
+    @asyncio.coroutine
+    def test_cursor_double_each(self):
+        range_size = 10000
+        cursor = yield from r.range().limit(range_size).run(self.conn)
+        count = 0
+
+        while (yield from cursor.fetch_next()):
+            yield from cursor.next()
+            count += 1
+        self.assertEqual(count, range_size,
+             "Expected %d results on the cursor, but got %d" % (range_size, count))
+
+        while (yield from cursor.fetch_next()):
+            yield from cursor.next()
+            count += 1
+        self.assertEqual(count, range_size,
+             "Expected no results on the second iteration of the cursor, but got %d" % (count - range_size))
+
+    # Used in wait tests
+    num_cursors=3
+
+    @asyncio.coroutine
+    def do_wait_test(self, wait_time):
+        cursors = [ ]
+        cursor_counts = [ ]
+        cursor_timeouts = [ ]
+        for i in xrange(self.num_cursors):
+            cur = yield from r.range().map(r.js("(function (row) {" +
+                                                "end = new Date(new Date().getTime() + 2);" +
+                                                "while (new Date() < end) { }" +
+                                                "return row;" +
+                                           "})")).run(self.conn, max_batch_rows=2)
+            cursors.append(cur)
+            cursor_counts.append(0)
+            cursor_timeouts.append(0)
+
+        @asyncio.coroutine
+        def get_next(cursor_index):
+            try:
+                if wait_time is None: # Special case to use the default
+                    yield from cursors[cursor_index].next()
+                else:
+                    yield from cursors[cursor_index].next(wait=wait_time)
+                cursor_counts[cursor_index] += 1
+            except r.RqlTimeoutError:
+                cursor_timeouts[cursor_index] += 1
+
+        # We need to get ahead of pre-fetching for this to get the error we want
+        while sum(cursor_counts) < 4000:
+            for cursor_index in xrange(self.num_cursors):
+                for read_count in xrange(random.randint(0, 10)):
+                    yield from get_next(cursor_index)
+
+        [cursor.close() for cursor in cursors]
+
+        return (sum(cursor_counts), sum(cursor_timeouts))
+
+    # @asyncio.coroutine
```

Comment:
Is there a reason these tests are commented out?  Were they taking too long?  If I remember correctly they're the longest-running tests in the suite.